refaster-support/src/test/java/tech/picnic/errorprone/refaster/matchers/IsLikelyTrivialComputationTest.java

```diff
@@ -0,0 +1,147 @@
+package tech.picnic.errorprone.refaster.matchers;
+
+import static com.google.errorprone.BugPattern.SeverityLevel.ERROR;
+
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.CompilationTestHelper;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.sun.source.tree.ReturnTree;
+import org.junit.jupiter.api.Test;
+
+final class IsLikelyTrivialComputationTest {
+  @Test
+  void matches() {
+    CompilationTestHelper.newInstance(MatcherTestChecker.class, getClass())
+        .addSourceLines(
+            "A.java",
+            "import java.util.function.Predicate;",
+            "",
+            "class A {",
+            "  String negative1() {",
+            "    return String.valueOf(1);",
+            "  }",
+            "",
+            "  String negative2() {",
+            "    return toString().toString();",
+            "  }",
+            "",
+            "  String negative3() {",
+            "    return \"foo\" + toString();",
+            "  }",
+            "",
+            "  byte negative4() {",
+            "    return \"foo\".getBytes()[0];",
+            "  }",
+            "",
+            "  int negative5() {",
+            "    int[] arr = new int[0];",
+            "    return arr[hashCode()];",
+            "  }",
+            "",
+            "  int negative6() {",
+            "    return 1 * 2;",
```

Comment:
For the `BinaryTree` we can check if both left and right are an `LiteralTree` or `IdentifierTree` as start? Also fine with leaving this as is.

Reply:
Yeah, one could argue that this _is_ a trivial computation. But then maybe so is `4 * 60 * 60` ("four hours"), which still wouldn't be flagged. And since the matcher is already quite complex, I'm not sure how far to go for diminishing returns. Even this case I don't expect to be triggered too frequently (at least in our internal code base).